pkg/networkserver/grpc_gsns.go

```diff
@@ -200,20 +184,14 @@ outer:
 			"device_uid", unique.ID(ctx, dev.EndDeviceIdentifiers),
 			"full_f_cnt_up", fCnt,
 			"last_f_cnt_up", dev.matchedSession.LastFCntUp,
+			"mac_version", dev.matchedMACState.LoRaWANVersion,
 		))
 
-		resetsFCnt := false
-		if dev.GetMACSettings().GetResetsFCnt() != nil {
-			resetsFCnt = dev.MACSettings.ResetsFCnt.Value
-		} else if ns.defaultMACSettings.GetResetsFCnt() != nil {
-			resetsFCnt = ns.defaultMACSettings.ResetsFCnt.Value
-		}
-
 		gap := uint32(math.MaxUint32)
 		if fCnt == 0 && dev.matchedSession.LastFCntUp == 0 &&
 			(len(dev.RecentUplinks) == 0 || dev.PendingSession != nil) {
 			gap = 0
-		} else if !resetsFCnt {
+		} else if !resetsFCnt(dev.EndDevice, ns.defaultMACSettings) {
```

Comment:
Why not consider pending MAC state here? `resetsFCnt()` doesn't do that, while it is unknown which `matchedSession` is dealt with here

Reply:
Whether or not device resets the `FCnt` is independent from either MAC state or the session, what do you mean - what would the pending MAC state be used for here?

Reply:
Ah, apologies, I see. There's actually no reason to pass the end device; `dev.MACSettings` would do too. Both are fine with me